docs/BRIDGE_FLOW.md

```diff
@@ -0,0 +1,304 @@
+1. The deposit scenario: a L1 smart contract can send a message to another L2 smart contract upon receiving a deposit of tokens. The L2 smart contract will receive, process and validate this message, and mint an equivalent amount of tokens in the L2
+2. The withdrawal scenario: the holder of L2 tokens can burn them, and in doing so, generate a message from the L2 smart contract to the L1 smart contract that held bridged tokens. The L1 smart contract can process the message and release the L1 tokens back to an address, defined in the message generated in the L2 by the user, and presumably controlled by this same user.
+
+In the case of the deposit scenario, the message is relayed from the L1 to the L2 by the validator set of the Fuel blockchain. The withdrawal messages, on the other hand, are indirectly included in the commits that assert the finality of the Fuel blockchain in the L1, which means that once the history of the Fuel ledger is considered final, immutable and anchored to the L1, any user can execute withdrawal messages in the L1 blockchain.
+
+### Block committing
+
+The Fuel blockchain aims to be scalable, and thus is able to generate blocks at a faster rate than other blockchains, without compromising on its security. It does so by inheriting the security of the L1 it is anchored to. The mechanism allowing this is an hybrid optimistic-ZK rollup (WIP). Blocks generated in the Fuel blockchain are grouped in epochs (e.g. an epoch can group together 10800 blocks), and epochs are identified and summarized in the last block of said epoch. This block can be committed to the L1 blockchain with a transaction to the `FuelChainState` contract, and will be accepted optimistically (i.e. the commit is considered valid unless proven otherwise - validation of the blocks are not considered in the scope of this document). The commits will be registered with the timestamp of the L1 blockchain. Then, once a certain time period has passed, the commit (along with the block) will be considered final. 
+
+![Block Committing](block_committing.png)
+
+### L1 Deposit
+
+The deposit process is straightforward: the user will send a transaction to deposit an asset in a L1 bridge contract, which will in turn emit a message through the `FuelMessagePortal` to be picked up by the Fuel blockchain 's validator set. These validators will make sure that the message is included in the next blocks of the Fuel blockchain. Message inclusion also carries the possibility of said message being processed by its recipient - in the case of L1 deposit, a message will be delivered from the L1 bridge contract to the L2 bridge contract. This last contract will have the capability of processing these messages, validating that their sender is the L1 bridge contract, and then proceeding to mint an equivalent asset and amount in the Fuel blockchain to the one originally depositted in the L1. Then, the minted assets become available to the user in the L2.
```

Comment:
It might be good to clarify that validators simply include messages into the utxo set, and it's up to the user or a third-party service to deliver that message to the appropriate contract. Delivering the message to the contract is enforced by the predicate on the message, so anyone on the chain can submit a tx to process it.